
syntax = "proto3";

package profiler.proto;
option java_package = "com.android.tools.profiler.proto";
option java_outer_classname = "GpuProfiler";

import "profiler.proto";

service GpuService {
  rpc GetData(GpuDataRequest) returns (GpuDataResponse) {}
//  rpc StartMonitoringApp(GpuStartRequest) returns (GpuStartResponse) {}
//  rpc StopMonitoringApp(GpuStopRequest) returns (GpuStopResponse) {}
}

message GpuDataRequest {
  string app_package_name = 1;
  string activity_name = 2;
  Session session = 3;
}

message GpuDataResponse {
  GpuData data = 1;
}

message GpuData {
  CommonData basic_info = 1;
  float      utilization = 2;
}
